lib/consumer/amqp_data_consumer.ex

```diff
@@ -116,6 +117,12 @@ defmodule Mississippi.Consumer.AMQPDataConsumer do
     end
   end
 
+  def handle_info({:EXIT, _from, {:name_conflict, {_key, _value}, _registry, _pid}}, state) do
+    _ = Logger.warning("Duplicate AMQPDataConsumer shutting down")
+    # Exit with :normal so we're not restarted
+    {:stop, :normal, state}
+  end
+
```

Comment:
Since we're now trapping exits, we should also handle other exit codes, otherwise we will get a crash even in case of a normal exit

```elixir
  def handle_info({:EXIT, _from, reason}, state) do
    {:stop, reason, state}
  end
```